core/src/main/java/org/openstack4j/openstack/identity/domain/v3/AccessWrapper.java

```diff
@@ -63,17 +64,40 @@ public static AccessWrapper wrap(KeystoneToken token) {
      */
     @Override
     public org.openstack4j.model.identity.Token getToken() {
-        Project project = KeystoneProject.builder()
-                .id(token.getProject().getId())
-                .name(token.getProject().getName())
-                .build();
-        Tenant tenant = KeystoneTenant.builder()
-                .id(project.getId())
-                .name(project.getName())
-                .description(project.getDescription())
-                .enabled(project.isEnabled())
-                .build();
-        return new V2Token(id, token.getExpires(), token.getVersion(), tenant);
+        /*This is only a temporary solution to get authentication with project and domain scope working for now.
+         * Proper implementation coming soon.
+         */
+        if(token.getProject() != null) {
+            Project project = KeystoneProject.builder()
+                    .id(token.getProject().getId())
+                    .name(token.getProject().getName())
+                    .build();
+            Tenant tenant = KeystoneTenant.builder()
+                    .id(project.getId())
+                    .name(project.getName())
+                    .description(project.getDescription())
+                    .enabled(project.isEnabled())
+                    .build();
+
+            return new V2Token(id, token.getExpires(), token.getVersion(), tenant);
+        }
+        else if(token.getDomain() != null) {
```

Comment:
Right. But if you don't mind I would leave it that way for now and do better on the new dedicated v3 branch :)